internal/payloads/reservation_payload.go

```diff
@@ -2,38 +2,77 @@ package payloads
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/RHEnVision/provisioning-backend/internal/models"
 	"github.com/go-chi/render"
 )
 
 // ReservationRequest is empty, account comes in HTTP header and
 // provider type in HTTP URL. All other fields are auto-generated.
-type ReservationRequest struct {
-}
 
-type ReservationResponse struct {
-	*models.Reservation
-}
+type ReservationResponsePayload struct {
+	ID int64 `json:"id"`
 
-type NoopReservationRequest ReservationRequest
+	// Provider type. Required.
+	Provider int `json:"provider"`
 
-type NoopReservationResponse struct {
-	*models.NoopReservation
+	// Time when reservation was made.
+	CreatedAt time.Time `json:"created_at"`
+
+	// Textual status of the reservation or error when there was a failure
+	Status string `json:"status"`
+
+	// Time when reservation was finished or nil when it's still processing.
+	FinishedAt *time.Time `json:"finished_at"`
+
+	// Flag indicating success, error or unknown state (NULL). See Status for the actual error.
+	Success *bool `json:"success"`
 }
 
-type AWSReservationRequest struct {
+type AWSReservationResponsePayload struct {
+	ReservationResponsePayload
 	*models.AWSReservation
```

Comment:
This line breaks it, instead of struct embedding just copy and paste the fields from the model:

```
	// Pubkey ID.
	PubkeyID int64 `db:"pubkey_id" json:"pubkey_id"`

	// Source ID.
	SourceID int64 `db:"source_id" json:"source_id"`

	//AWS Instance type.
	InstanceType string `db:"instance_type" json:"instance_type"`

	// Amount of instances to provision of type: Instance type.
	Amount int32 `db:"amount" json:"amount"`

	// The ID of the image from which the instance is created.
	ImageID string `db:"image_id" json:"image_id"`

	// The ID of the aws reservation which was created.
	AWSReservationID string `db:"aws_reservation_id" json:"aws_reservation_id"`
```

Make sure to remove all the db tags from here, and in the model remove json tags.

Alternatively, if you still want to use embedding, you need to split the model into two structs and only embed the detail part:

```go
type AWSReservationDetail struct {
	// Pubkey ID.
	PubkeyID int64 `db:"pubkey_id" json:"pubkey_id"`

	// Source ID.
	SourceID int64 `db:"source_id" json:"source_id"`

	//AWS Instance type.
	InstanceType string `db:"instance_type" json:"instance_type"`

	// Amount of instances to provision of type: Instance type.
	Amount int32 `db:"amount" json:"amount"`

	// The ID of the image from which the instance is created.
	ImageID string `db:"image_id" json:"image_id"`

	// The ID of the aws reservation which was created.
	AWSReservationID string `db:"aws_reservation_id" json:"aws_reservation_id"`
}

type AWSReservation struct {
	Reservation
	AWSReservationDetail
}
```

In that case keep both db and json tags. You can only use this when there are no NULLable fields tho. It is up to you how to approach it, I think the split should be better.